service/resource/deployment/resource.go

```diff
@@ -121,6 +121,14 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 			return nil, microerror.Mask(err)
 		}
 		deployments = append(deployments, workerDeployments...)
+
+		if key.HasNodeController(customObject) {
+			nodeControllerDeployment, err := newNodeControllerDeployment(customObject)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			deployments = append(deployments, nodeControllerDeployment)
+		}
```

Comment:
FYI IMO this is a hack we will stop doing quite soon. The implication of adding such functionality will soon be to create a new version bundle to create a new minor release of the kvm-operator. We would then copy the deployment resource (or all maybe all?) and route custom objects within the operatorkit framework accordingly. Lets please create an issue to track the cleanup of this code. At some point all custom objects should have the node controller configuration and we can remove the condition.

Reply:
Here we go https://github.com/giantswarm/kvm-operator/issues/223